WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java

```diff
@@ -433,22 +435,28 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String permissi
                 if (permissions == null || permissions.length == 0) {
                     AnalyticsTracker.track(AnalyticsTracker.Stat.ME_GRAVATAR_PERMISSIONS_INTERRUPTED);
                 }  else {
-                    for (int i = 0; i < grantResults.length; i++) {
-                        if (grantResults[i] == PackageManager.PERMISSION_DENIED) {
-                            ToastUtils.showToast(this.getActivity(), getString(R.string
-                                    .gravatar_camera_and_media_permission_required), ToastUtils.Duration.LONG);
-
-                            Map<String, Object> properties = new HashMap<>();
-                            properties.put("permission denied", permissions[i]);
-                            AnalyticsTracker.track(AnalyticsTracker.Stat.ME_GRAVATAR_PERMISSIONS_DENIED, properties);
+                    List<String> granted = new ArrayList<>();
+                    List<String> denied = new ArrayList<>();
 
-                            return;
+                    for (int i = 0; i < grantResults.length; i++) {
+                        if (grantResults[i] == PackageManager.PERMISSION_GRANTED) {
+                            granted.add(permissions[i]);
+                        } else {
+                            denied.add(permissions[i]);
                         }
                     }
 
-                    AnalyticsTracker.track(AnalyticsTracker.Stat.ME_GRAVATAR_PERMISSIONS_ACCEPTED);
-
-                    askForCameraOrGallery();
+                    if (denied.size() == 0) {
+                        AnalyticsTracker.track(AnalyticsTracker.Stat.ME_GRAVATAR_PERMISSIONS_ACCEPTED);
+                        askForCameraOrGallery();
+                    } else {
+                        ToastUtils.showToast(this.getActivity(), getString(R.string
+                                .gravatar_camera_and_media_permission_required), ToastUtils.Duration.LONG);
+                        Map<String, Object> properties = new HashMap<>();
+                        properties.put("permissions granted", TextUtils.join(",", granted));
```

Comment:
I did want it there for clarity. People seeing in the Analytics interface might be confused by its absence. Having said that, I will make it more clear though by putting a "none" text there in the case of empty list so the reader will know it's empty. Thanks!